pkg/scale/encode.go

```diff
@@ -223,6 +225,35 @@ func (es *encodeState) encodeArray(in interface{}) (err error) {
 	return
 }
 
+func (es *encodeState) encodeMap(in interface{}) (err error) {
+	v := reflect.ValueOf(in)
+	err = es.encodeLength(v.Len())
+	if err != nil {
+		return
+	}
+
+	for i := v.MapRange(); i.Next(); {
+		fmt.Println(i.Key(), "\t:", i.Value())
+
+		key := i.Key()
+		err = es.marshal(key.Interface())
+		if err != nil {
+			return
+		}
+
+		mapValue := i.Value()
+		if !mapValue.CanInterface() {
+			continue
+		}
```

Comment:
Shouldn't we return an error in that case, to avoid silently discarding a map value when encoding?

@timwu20

Reply:
Def wait for Tims answer, but this happens at other places in this file as well (i.e. structs)

Reply:
'CanInterface` is essentially determining if this is a public attribute of a struct or public method.  I think it should be fine in this case.  This will almost always return true in the case we're trying to decode into a map value.  I wonder what cases this would return false though, maybe we can provide a test case.

@kishansagathiya please don't merge the PR with unresolved conversations.  My bad for not getting to this earlier.

Reply:
I realised about this comment few moments after merging it. Wasn't deliberate.